salt/config/__init__.py

```diff
@@ -3563,7 +3613,7 @@ def apply_master_config(overrides=None, defaults=None):
         opts['ipc_write_buffer'] = 0
     using_ip_for_id = False
     append_master = False
-    if not opts.get('id'):
+    if not opts.get('id') or opts.get('minion_id_type', 'str') == 'func':
```

Comment:
@isbm : I'll just wait for @cachedout to comment on the design and approach before modifying things. This way, I will know if the fundamental design should be changed.